Feature #1885 gh_projects

JohnHalleyGotway opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Just wrote the docs as I created the new METplus-Analysis-5.1.0 Development project.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- @fisherhucar please review the newly created [METplus-Analysis-5.1.0 Development](https://github.com/orgs/dtcenter/projects/68) project contents.
- @jprestop please review the new GitHub workflow section [GitHub Projects to manage development](https://metplus.readthedocs.io/en/feature_1885_gh_projects/Contributors_Guide/github_workflow.html#github-projects-to-manage-development).
   - Julie, I did see the other proposed edits listed in the issue (fixing a broken link, changing the affiliation for James), but I couldn't actually find anything that needed to be fixed regarding that. So perhaps those changes have already been made?

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**
These are doc only updates.

- [x] Do these changes include sufficient testing updates? **[No]**
None needed.

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[Fri 12/2/22]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

JohnHalleyGotway committed:
Per #1885, add instructions for creating development projects.

JohnHalleyGotway committed:
Per #1885, add instructions for creating development projects.

JohnHalleyGotway committed:
Per #1885, insert missing space.

JohnHalleyGotway committed:
Per #1885, add detail about project workflows.

JohnHalleyGotway committed:
Per #1885, more bolding.

jprestop commented on docs/Contributors_Guide/github_workflow.rst:
```diff
@@ -23,6 +23,140 @@ then the develop branch is used to create a main_vx.y release of METplus,
+        the **X** to its right.
+
+      - Select **Iteration** and modify the **Field name** to be **Cycle**.  Delete any existing cycles.
+        Under **More options** select a **Start on* date and set the default **Duration** as 6 weeks.
```
```suggestion
        Under **More options** select a **Start on** date and set the default **Duration** as 6 weeks.
```

jprestop commented on docs/Contributors_Guide/github_workflow.rst:
```diff
@@ -23,6 +23,140 @@ then the develop branch is used to create a main_vx.y release of METplus,
+6. Link the new project to each repository.
+
+   - Navigate to the project page for each repository managed by this project
+     (e.g. `METplus Projects <https://github.com/dtcenter/METviewer/projects>`_).
```
This says "METplus Projects", but links to METviewer projects.  I am guessing that wasn't intended, but I'm not sure if you'd rather say "METviewer projects" or link to METplus or if you have no preference.  Please fix however you'd like.

jprestop reviewed (approved):
This is amazing!  Thank you!  I'd be happy to run through these steps in setting up a future project for the next release.  I made a couple of suggested changes:
1. Fix formatting for some bolded text
2. Fix either text or a link

Once those changes are made I approve this request.  I'll go ahead and approve now, instead of "Request changes" since they're minor changes and then you won't have to wait for me to approve in the future.

JohnHalleyGotway committed:
Per #1885, add section about support projects.

JohnHalleyGotway committed:
Per #1885, update Release guide to link to the contributors guide details.

JohnHalleyGotway commented:
- Added a new section named **GitHub Projects to manage support** and created a new [METplus Version 5.0 Support](https://github.com/orgs/dtcenter/projects/69) project.
- Just pushed changes to update the release guide slightly and add links to the contributor's guide.
- Not sure if I found all spots where changes are needed, but we can find/fix more later.
- Adding @georgemccabe as a reviewer to get his perspective.

JohnHalleyGotway committed:
Per #1885, use consist spelling of reassign.

jprestop commented:
I went ahead and re-reviewed for the Release Notes changes.  It looks good to me @JohnHalleyGotway.  Thanks for realizing there were changes needed there as well.

georgemccabe commented on docs/Contributors_Guide/github_workflow.rst:
```diff
@@ -22,6 +22,269 @@ develop branch.  When a release candidate for METplus has been determined,
+      - View name **Closed Issues** shows issues that have been closed across all development cycles.
+        Click on the 3 horizontal bars and define the filtering criteria as **is:closed is:issue**.
+        Click on the 3 dots in the **Cycle** column and select **Group by values**.  Click **Save Changes**.
+
```
The METplus project boards have included another view that I find useful. This isn't mandatory to add.

```suggestion

      - View name **High/Blocker Not Required** shows all items labelled as *HIGH* or *BLOCKER* priority but
        not marked as required for the development or official release.
        Click on the 3 horizontal bars and define the filtering criteria as
        **is:open label:"priority: high","priority: blocker" -label:"required: FOR DEVELOPMENT RELEASE" -label:"required: FOR OFFICIAL RELEASE"**.
        Click **Save Changes**.

```

georgemccabe commented on docs/Contributors_Guide/github_workflow.rst:
```diff
@@ -22,6 +22,269 @@ develop branch.  When a release candidate for METplus has been determined,
+      - The blue dot on the down arrow for this tab indicates that there are unsaved changes.
+        Select the **Save changes** option.
+
+   - Click on the down arrow and select **Duplicate View**.  Do this 7 times and name/refine these views as follows.
```
If the suggestion below this one is applied, then the number here should be updated.

```suggestion
   - Click on the down arrow and select **Duplicate View**.  Do this 8 times and name/refine these views as follows.
```

georgemccabe reviewed (approved):
This content looks great, thank you for adding it. There is another label that is used for METplus that I find useful. The suggested changes include instructions to create it. If you don't think this is necessary, then feel free to ignore the suggestion. I approve either way.

JohnHalleyGotway committed:
Update docs/Contributors_Guide/github_workflow.rst

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

JohnHalleyGotway committed:
Update docs/Contributors_Guide/github_workflow.rst

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

JohnHalleyGotway commented:
@jprestop and @georgemccabe, thanks for the reviews and suggestions. George, I added your recommended view.

I just removed @fisherhucar from the reviewers since 2 is certainly plenty.

Merging now.
